src/components/logical-expression-builder/logical-expression-builder.component.ts

```diff
@@ -0,0 +1,149 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
+import { LogicalOperatorDefinition } from './interfaces/LogicalOperatorDefinition';
+import { OperatorDefinitionList } from './interfaces/OperatorDefinitionList';
+import { FieldDefinition } from './interfaces/FieldDefinition';
+import { LogicalExpressionBuilderService } from './services/logical-expression-builder.service';
+import { LogicalExpressionBuilderExpression, ExpressionCondition } from './interfaces/LogicalExpressionBuilderExpression';
+import { DisplayValueFunction } from './interfaces/DisplayValueFunction';
+
+@Component({
+    selector: 'ux-logical-expression-builder',
+    templateUrl: './logical-expression-builder.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class LogicalExpressionBuilderComponent {
+
+    @Input()
+    set logicalOperators(logicalOperators: LogicalOperatorDefinition[]) {
+        this._lebService.setLogicalOperators(logicalOperators);
+    }
+
+    @Input()
+    set operators(operators: OperatorDefinitionList) {
+        this._lebService.setOperators(operators);
+    }
+
+    @Input()
+    set fields(fields: FieldDefinition[]) {
+        this._lebService.setFields(fields);
+    }
+
+    @Input()
+    set expression(expression: LogicalExpressionBuilderExpression) {
+        this._expression = this.addEditableFieldToConditionsInExpression(expression);
+        this.expressionChange.emit(this.cleanExpression(this._expression));
+    }
+
+    get expression() {
+        return this._expression;
+    }
+
+    @Input()
+    set localizedStrings(localizedStrings: any) {
+        this._lebService.setLocalizedStrings(localizedStrings);
+    }
+
+    @Input()
+    set displayValueFunction(displayValueFunction: DisplayValueFunction) {
+        this._lebService.setDisplayValueFunction(displayValueFunction);
+    }
+
+    private _expression: LogicalExpressionBuilderExpression;
+
+    @Output() expressionChange = new EventEmitter<LogicalExpressionBuilderExpression>();
+
+    constructor(private _lebService: LogicalExpressionBuilderService) {
+    }
+
+    public isExpressionEmpty(): boolean {
+        return !this.expression?.type;
+    }
+
+    public getLogicalOperatorName(): string {
+        return ('logicalOperator' in this.expression) ? this.expression.logicalOperator : null;
+    }
+
+    public handleGroupChange(expression: LogicalExpressionBuilderExpression) {
+        let temp = { ...expression };
+
+        // make expression just a condition if it contains exactly one group with exactly one condition in it
+        if (temp?.type === 'group'
+            && ('children' in temp)
```

Comment:
```suggestion
            && (temp.hasOwnProperty('children'))
```
However, this is already covered in the next line.